packages/svelte/src/internal/client/dev/elements.js

```diff
@@ -1,6 +1,20 @@
 import { HYDRATION_END, HYDRATION_START } from '../../../constants.js';
-import { hydrating } from '../dom/hydration.js';
-import { is_array } from '../utils.js';
+import { hydrate_end, hydrate_start, hydrating } from '../dom/hydration.js';
+
+/**
+ * @param {import('#client').TemplateNode} from
+ * @param {import('#client').TemplateNode} to
+ */
+function find_nodes_between(from, to) {
```

Comment:
After doing a deep dive into before and after, the performance loss seems to be exclusively coming from this function and `remove_nodes` and the fact both trigger `nextSibling`. In fact, the calls to `nextSibling` before take 34ms for me, and with this PR, that goes to 51ms. I wonder if we can somehow avoid this again.

Reply:
This is dev-only, it's for `add_locations`